internal/cmd/base/servers.go

```diff
@@ -548,20 +549,37 @@ func (b *Server) RunShutdownFuncs() error {
 }
 
 func (b *Server) ConnectToDatabase(dialect string) error {
-	dbase, err := gorm.Open(dialect, b.DatabaseUrl)
+	dbType, err := db.StringToDbType(dialect)
+	if err != nil {
+		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
+	}
+	dbase, err := db.Open(dbType, b.DatabaseUrl)
 	if err != nil {
 		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
 	}
 	if b.DatabaseMaxOpenConnections == 1 {
 		return fmt.Errorf("unable to create db object with dialect %s: %s", dialect, "max_open_connections must be unlimited by setting 0 or at least 2")
 	} else {
-		dbase.DB().SetMaxOpenConns(b.DatabaseMaxOpenConnections)
+		underlyingDB, err := dbase.DB()
+		if err != nil {
+			return fmt.Errorf("unable retreive db: %w", err)
+		}
+		underlyingDB.SetMaxOpenConns(b.DatabaseMaxOpenConnections)
 	}
 	b.Database = dbase
```

Comment:
this line can be removed since `b.Database = dbase` is being set after the `if` and `b.Database` is not used in that `if` block.

Reply:
see:  https://github.com/hashicorp/boundary/pull/1535